ivy/functional/frontends/numpy/indexing_routines/inserting_data_into_arrays.py

```diff
@@ -28,3 +30,138 @@ def fill_diagonal(a, val, wrap=False):
+            elif isinstance(item, str):
+                if k != 0:
+                    raise ValueError("special directives must be the first entry.")
+                if item in ("r", "c"):
+                    matrix = True
+                    col = item == "c"
+                    continue
+                if "," in item:
+                    vec = item.split(",")
+                    try:
+                        axis, ndmin = [int(x) for x in vec[:2]]
+                        if len(vec) == 3:
+                            trans1d = int(vec[2])
+                        continue
+                    except Exception as e:
+                        raise ValueError(
+                            "unknown special directive {!r}".format(item)
+                        ) from e
+                try:
+                    axis = int(item)
+                    continue
+                except (ValueError, TypeError) as e:
+                    raise ValueError("unknown special directive") from e
+            elif (ivy.isscalar(item)) or (ivy.is_ivy_array(item) and item.ndim == 0):
+                scalar = True
+                newobj = item
+            else:
+                item = ivy.array(item)
+                item_ndim = item.ndim
+                newobj = np_frontend.array(item, copy=False, ndmin=ndmin).ivy_array
+                if trans1d != -1 and item_ndim < ndmin:
+                    k2 = ndmin - item_ndim
+                    k1 = trans1d
+                    if k1 < 0:
+                        k1 += k2 + 1
+                    defaxes = list(range(ndmin))
+                    axes = defaxes[:k1] + defaxes[k2:] + defaxes[k1:k2]
+                    newobj = np_frontend.transpose(newobj, axes=axes).ivy_array
+
+            objs.append(newobj)
+            if scalar:
+                result_type_objs.append(item)
+            else:
+                result_type_objs.append(newobj.dtype)
+
+        # Ensure that scalars won't up-cast unless warranted, for 0, drops
+        # through to error in concatenate.
+        if len(result_type_objs) != 0:
+            if len(result_type_objs) > 1:
+                final_dtype = ivy.result_type(*result_type_objs)
+            else:
+                final_dtype = ivy.result_type(result_type_objs[0], result_type_objs[0])
+            # concatenate could do cast, but that can be overridden:
+            objs = [
+                np_frontend.array(
+                    obj, copy=False, ndmin=ndmin, dtype=final_dtype
+                ).ivy_array
+                for obj in objs
+            ]
+
+        res = self.concatenate(tuple(objs), axis=axis)
+
+        if matrix:
+            oldndim = res.ndim
+            res = self.makemat(res)
+            if oldndim == 1 and col:
+                res = res.T
+        return res
+
+    def __len__(self):
+        return 0
+
+
+class RClass(AxisConcatenator):
+    def __init__(self):
+        AxisConcatenator.__init__(self, 0)
```

Comment:
We can use `super().__init__()` to conform with best inheritance pratices

Reply:
yes right, done!